telethon/client/chats.py

```diff
@@ -1145,6 +1145,23 @@ async def kick_participant(
         else:
             raise ValueError('You must pass either a channel or a chat')
 
+    async def get_permissions(
+            self: 'TelegramClient',
+            channel: 'hints.EntityLike',
+            user: 'hints.EntityLike'
+    ) -> 'custom.ParticipantPermissions':
+        channel = await self.get_input_entity(channel)
+        user = await self.get_input_entity(user)
+        if helpers._entity_type(user) != helpers._EntityType.USER:
+            raise ValueError('You must pass a user entity')
+        if helpers._entity_type(channel) != helpers._EntityType.CHANNEL:
+            raise ValueError('You must pass a channel entity')
+        participant = await self(functions.channels.GetParticipantRequest(
+            channel,
+            user
+        ))
+        return custom.ParticipantPermissions(participant.participant)
```

Comment:
```suggestion
    if helpers._entity_type(channel) == helpers._EntityType.CHANNEL:
        participant = await self(functions.channels.GetParticipantRequest(
            entity,
            user
        ))
        participant = participant.participant
    else:
        participants = (await client(functions.messages.GetFullChatRequest(chat_id))) \
            .full_chat.participants.participants
        participant = next((participant for participant in participants if participant.user_id == user_id), None)
        if not participant:
            raise ValueError('User must be participant of Chat')
    return custom.ParticipantPermissions(participant)
```

Reply:
Getting participant is different for chat and channel. Also raise `ValueError`, like `GetParticipantRequest` would raise an rpc error if participant not found.

Reply:
Getting participant is different for chat and channel. Also raise ValueError, like GetParticipantRequest would raise an rpc error if participant not found.